lib/src/main/java/com/cloudbees/groovy/cps/impl/CpsBooleanClosureWrapper.java

```diff
@@ -0,0 +1,43 @@
+package com.cloudbees.groovy.cps.impl;
+
+import groovy.lang.Closure;
+import org.codehaus.groovy.runtime.callsite.BooleanReturningMethodInvoker;
+
+import java.io.Serializable;
+import java.util.Map;
+
+/**
+ * A serializable equivalent of {@link org.codehaus.groovy.runtime.callsite.BooleanClosureWrapper}, where the
+ * {@link BooleanReturningMethodInvoker} is instantiated when {@link #call(Object...)} is called to avoid serialization
+ * issues with that as well.
+ */
+public class CpsBooleanClosureWrapper implements Serializable {
+    private final Closure wrapped;
+    private final int numberOfArguments;
+
+    public CpsBooleanClosureWrapper(Closure wrapped) {
+        this.wrapped = wrapped;
+        numberOfArguments = wrapped.getMaximumNumberOfParameters();
+    }
+
+    /**
+     * normal closure call
+     */
+    public boolean call(Object... args) {
+        BooleanReturningMethodInvoker bmi = new BooleanReturningMethodInvoker("call");
+        return bmi.invoke(wrapped, args);
+    }
+
+    /**
+     * Bridge for a call based on a map entry. If the call is done on a {@link Closure}
+     * taking one argument, then we give in the {@link Map.Entry}, otherwise we will
+     * give in the key and value.
+     */
+    public <K,V> boolean callForMap(Map.Entry<K, V> entry) {
+        if (numberOfArguments==2) {
+            return call(entry.getKey(), entry.getValue());
+        } else {
+            return call(entry);
```

Comment:
Maybe just create a new `BooleanClosureWraper` and delegate to it, so we have less code here?

Reply:
Good point.